readme.md

```diff
@@ -71,7 +71,11 @@ There are two parts to creating an Application Link. The first is done in Jenkin
 To register a consumer: 
 1. In Jenkins go to **Jenkins** > **Manage Jenkins** > **Manage Bitbucket Server consumers**. 
 2. Select **Register new consumer**. 
-3. Enter consumer details. 
+3. Enter these details: 
+   - Consumer key: A unique identifier for your Bitbucket instance. 
+   - Consumer name: A human-readable identifier for your Bitbucket instance.  
+   - Consumer secret: A password Jenkins can use with your key to identify your Bitbucket instance. Once this is entered you won't be able to retrieve it. 
+   - Callback URL (optional): The address Jenkins should redirect to after an OAuth flow. For example, your Bitbucket instance's base URL. 
```

Comment:
Can you please describe a bit more about why its optional? Perhaps we can say something like "If you leave this blank, it will default to...."

Reply:
I had a chat with Gaurav about this, and I think we should _not_ refer to the callback URL as optional. Although at present it at best covers a corner case, which I think is too technical to go into how-to documentation, we may in the future need to use it for request validation